### ISSUE TITLE
Navigator navigationBar renders at bottom of screen

### ISSUE THREAD

**Issue by esauter5** [facebook/react-native#1278]
I just implemented a `Navigator` component with a custom `navigationBar`. When it renders, the navBar renders at the bottom of the screen, but otherwise renders correctly. I noticed that this is because in the code, navBar renders after the scene: https://github.com/facebook/react-native/blob/master/Libraries/CustomComponents/Navigator/Navigator.js#L1318-L1321

I noticed that `Navigator.NavigationBar` gets around this by setting the navBar component to `position: absolute` (https://github.com/facebook/react-native/blob/master/Libraries/CustomComponents/Navigator/NavigatorNavigationBar.js#L190-L195)

Would it make sense to have the navbar render first so that it was at the top by default? I can turn open a PR if so.

**Comment by jmstout** [comment]
I would imagine part of the reason the navbar is rendered after the scene is so it appears on top of the scene, rather than below it.

As far as I understand it, the component order is basically equivalent to z-index in css.

**Comment by esauter5** [comment]
Thanks @jmstout. If this is true, then it makes sense. I realized the flaw in my logic: I was pondering the navbar being on top for a screen that didnt scroll so it seemed natural. If the screen did scroll, the navbar would go away unless you have the absolute positioning. Not sure how I overlooked that. Going to close.

**Comment by mattvagni** [comment]
📢  For anyone else ending up on this *still* (like me) an easy way to reverse the order is to simply add `flexDirection: 'column-reverse'` to the Navigator.

**Comment by iamwebdesigner** [comment]
This is a little bit crazy. I was implementing custom `navBar` component and was kind of surprised to see it at the bottom of the page. I tried some stuff advised here and in other resources, but none of them helped, so I started to digging the source code of this and dependent libraries. After 2 hours of mind-blowing work, I ran against this piece of code on https://github.com/aksonov/react-native-experimental-navigation/blob/master/NavigationAnimatedView.js#L165
```jsx
    return (
      <View
        onLayout={this._onLayout}
        style={this.props.style}>
        <View style={styles.scenes} key="scenes">
          {scenes}
        </View>
        {overlay} <-- renders navBar
      </View>
    );
```
I kind of felt some dirty stuff here, as `navBar` component is rendered after the scenes, so when I moved `{overlay}` above the `View` rendering the `scenes`, my custom `navBar` component finally got to the top.
I am new to `react-native` and do not understand how everything works here and I can't think this is a mistake, so I decided to put it here. I hope somebody would open the PR, if this is a solution for the problem.
